python/tests/test_tree_stats.py

```diff
@@ -5854,6 +5854,39 @@ def get_example_ts(self):
         assert ts.num_sites > 1
         return ts
 
+    def test_one_way_no_window_scalar_stat(self):
+        ts = self.get_example_ts()
+        x = ts.diversity()
+        assert isinstance(x, np.floating)
+
+    def test_one_way_one_window_scalar_stat(self):
+        ts = self.get_example_ts()
+        x = ts.diversity(windows=[0, ts.sequence_length])
+        assert x.shape == (1,)
+
+    def test_multi_way_no_window_scalar_stat(self):
+        ts = self.get_example_ts()
+        n = ts.num_samples
+        x = ts.f2(
+            sample_sets=[
+                [i for i in range(0, int(n / 2))],
+                [i for i in range(int(n / 2), n)],
+            ]
+        )
+        assert isinstance(x, np.floating)
+
+    def test_multi_way_one_window_scalar_stat(self):
+        ts = self.get_example_ts()
+        n = ts.num_samples
+        x = ts.f2(
+            sample_sets=[
+                [i for i in range(0, int(n / 2))],
+                [i for i in range(int(n / 2), n)],
+            ],
+            windows=[0, ts.sequence_length],
+        )
+        assert x.shape == (1,)
+
```

Comment:
```suggestion

    def test_multi_way_indexes_scalar_stat(self):
        ts = self.get_example_ts()
        n = ts.num_samples
        x = ts.f2(
            sample_sets=[
                [i for i in range(0, int(n / 2))],
                [i for i in range(int(n / 2), n)],
            ],
            indexes=(0, 1),
        )
        assert x.shape == (1,)
```

Reply:
So with k-way stats, do we want to a drop to a scalar only if { windows=None } OR { there are k sample sets and no indices were provided} ?